From 53cdcb0e9429b636652cb128b23e1a9f4bb4a042 Mon Sep 17 00:00:00 2001
From: Aryan Khare <aryakhar@qti.qualcomm.com>
Date: Thu, 23 Oct 2025 12:12:18 +0530
Subject: [PATCH] Changed LOG Roll-Over to FALSE in beerocks_agent.conf
 Additional check also added to retain STATUS in long run

Signed-off-by: Aryan Khare <aryakhar@qti.qualcomm.com>
---
 CMakeLists.txt                                      | 2 +-
 common/beerocks/bcl/source/beerocks_config_file.cpp | 2 +-
 common/beerocks/bcl/source/beerocks_logging.cpp     | 2 +-
 common/beerocks/scripts/prplmesh_utils.sh.in        | 9 +++++++--
 4 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2dada9e81..9180294a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -216,7 +216,7 @@ set(BEEROCKS_BH_WIRE_IFACE "eth1" CACHE STRING "Backhaul wire iface")
 set(BEEROCKS_LOG_FILES_ENABLED   "true" CACHE STRING "Write prplMesh logs into files")
 set(BEEROCKS_LOG_FILES_PATH      "${TMP_PATH}/logs" CACHE PATH "prplMesh logs directory")
 set(BEEROCKS_LOG_FILES_SUFFIX    ".log" CACHE STRING "prplMesh logs filenames suffix")
-set(BEEROCKS_LOG_FILES_AUTO_ROLL "true" CACHE STRING "Auto rollback prplMesh logs")
+set(BEEROCKS_LOG_FILES_AUTO_ROLL "false" CACHE STRING "Auto rollback prplMesh logs")
 set(BEEROCKS_LOG_STDOUT_ENABLED  "false" CACHE STRING "Print logs to stdout")
 set(BEEROCKS_LOG_SYSLOG_ENABLED  "true" CACHE STRING "Send logs to syslog")
 if (TARGET_PLATFORM STREQUAL "linux")
diff --git a/common/beerocks/bcl/source/beerocks_config_file.cpp b/common/beerocks/bcl/source/beerocks_config_file.cpp
index e556768a3..4542055f5 100644
--- a/common/beerocks/bcl/source/beerocks_config_file.cpp
+++ b/common/beerocks/bcl/source/beerocks_config_file.cpp
@@ -26,7 +26,7 @@ static bool read_log_section(const std::string &config_file_path, config_file::S
         std::make_tuple("log_global_size=", &sLogConf.global_size, mandatory),
         std::make_tuple("log_files_enabled=", &sLogConf.files_enabled, mandatory),
         std::make_tuple("log_files_path=", &sLogConf.files_path, mandatory),
-        std::make_tuple("log_files_auto_roll=", &sLogConf.files_auto_roll, mandatory),
+        std::make_tuple("log_files_auto_roll=", &sLogConf.files_auto_roll, optional),
         std::make_tuple("log_stdout_enabled=", &sLogConf.stdout_enabled, mandatory),
         std::make_tuple("log_syslog_enabled=", &sLogConf.syslog_enabled, optional)};
 
diff --git a/common/beerocks/bcl/source/beerocks_logging.cpp b/common/beerocks/bcl/source/beerocks_logging.cpp
index 92c613170..75a548283 100644
--- a/common/beerocks/bcl/source/beerocks_logging.cpp
+++ b/common/beerocks/bcl/source/beerocks_logging.cpp
@@ -345,7 +345,7 @@ logging::logging(const std::string &module_name, const beerocks::config_file::SC
         {"log_files_enabled", settings.files_enabled.empty() ? "true" : settings.files_enabled});
     m_settings_map.insert({"log_files_path", settings.files_path});
     m_settings_map.insert({"log_files_auto_roll",
-                           settings.files_auto_roll.empty() ? "true" : settings.files_auto_roll});
+                           settings.files_auto_roll.empty() ? "false" : settings.files_auto_roll});
     m_settings_map.insert({"log_global_levels", settings.global_levels});
     m_settings_map.insert({"log_global_syslog_levels", settings.syslog_levels});
     m_settings_map.insert({"log_global_size", settings.global_size});
diff --git a/common/beerocks/scripts/prplmesh_utils.sh.in b/common/beerocks/scripts/prplmesh_utils.sh.in
index 1bfd31ee7..b70c2f03a 100755
--- a/common/beerocks/scripts/prplmesh_utils.sh.in
+++ b/common/beerocks/scripts/prplmesh_utils.sh.in
@@ -280,8 +280,13 @@ execute_beerocks_command() {
 # line ($) or blank character ([[:blank:]])
 
 main_agent_operational() {
-    pgrep -fl beerocks_agent  > /dev/null 2>&1 || return 1
-    grep -q ' --> OPERATIONAL' "$1/beerocks_backhaul.log"
+	pgrep -fl beerocks_agent > /dev/null 2>&1 || return 1
+	if grep -q ' --> OPERATIONAL' "$1/beerocks_backhaul@BEEROCKS_LOG_FILES_SUFFIX@"* && \
+		! grep -q ' --> Terminated' "$1/beerocks_backhaul@BEEROCKS_LOG_FILES_SUFFIX@"*; then
+			return 0
+	else
+		return 1
+	fi
 }
 
 radio_agent_operational() {
-- 
2.34.1

